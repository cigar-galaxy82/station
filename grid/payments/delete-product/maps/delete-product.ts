/* eslint-disable jest/no-export */

import { describe, expect } from '@jest/globals';
import { SuperfaceTest } from '@superfaceai/testing';

import { createProduct } from '../../create-product/maps/create-product';

export function deleteProductTest(providerName: string): void {
  describe(`payments/delete-product/${providerName}`, () => {
    let superface: SuperfaceTest;

    beforeEach(() => {
      superface = new SuperfaceTest({
        profile: 'payments/delete-product',
        provider: providerName,
        testInstance: expect,
      });
    });

    describe('DeleteProduct', () => {
      describe('when all inputs are correct', () => {
        it('deletes a product', async () => {
          const id = await createProduct(providerName);
          const result = await superface.run(
            {
              useCase: 'DeleteProduct',
              input: {
                id,
              },
            },
            {
              hideInput: ['id'],
            }
          );
          expect(() => result.unwrap()).not.toThrow();
          expect(result.unwrap()).toMatchSnapshot();
        });
      });
    });
  });
}
