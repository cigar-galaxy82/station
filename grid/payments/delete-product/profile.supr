"""
Delete Product
Deletes a product in a payment system (e.g. PayPal, Stripe...)
"""
name = "payments/delete-product"
version = "1.0.0"

"""
Delete Product
Deletes a product in a payment system.
"""

usecase DeleteProduct unsafe {
  input {
    """
    ID
    Unique identifier of the product to be deleted.
    """
    id! string!
  }

  error {
    """
    Title
    A short, human-readable summary of the problem type.
    """
    title! string

    """
    Detail
    A human-readable explanation specific to this occurrence of the problem.
    """
    detail string
  }

  example Successful {
    input {
      id = "PROD-1192810198"
    }
  }
}