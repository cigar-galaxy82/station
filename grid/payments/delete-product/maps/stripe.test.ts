import { deleteProductTest } from './delete-product';

deleteProductTest('stripe');
