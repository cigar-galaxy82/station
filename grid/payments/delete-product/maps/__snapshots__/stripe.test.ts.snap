// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`payments/delete-product/stripe DeleteProduct when all inputs are correct deletes a product 1`] = `undefined`;
